Public/Src/Cache/VerticalStore/BuildCacheAdapter/DistributedBuildCacheFactory.cs

```diff
@@ -160,5 +161,40 @@ private static BuildXL.Cache.MemoizationStore.Interfaces.Caches.ICache CreateDis
             ReadThroughMode readThroughMode = cacheConfig.SealUnbackedContentHashLists ? ReadThroughMode.ReadThrough : ReadThroughMode.None;
             return new DistributedCache(logger, innerCache, metadataCache, metadataTracer, readThroughMode);
         }
+
+        /// <inheritdoc />
+        public IEnumerable<Failure> ValidateConfiguration(ICacheConfigData cacheData)

```

Comment:
>ValidateConfiguration [](start = 36, length = 21)

I noticed quite a few overlaps between the implementations. If thye share a type should we have a helper method to validate the common cases?